### PR TITLE
Refactor of PG Connection/DB/TABLE Creation

### PR DESCRIPTION
**Pull request by dbkbali**
Refactor:

 - ensures connection pooling is optimized
 - removes db and table creation from shell scripts and into the pg connection interface
 - allows 1 place schema adjustments and regeneration simply by amending schema in 1 file ./shyft_schema/pg_table_schema.go
 - facilitates easier integration testing
- dockerfile configs and shell scripts have been modified accordingly

### CODE REVIEW

**Review by dutterbutter — COMMENTED** [review]
Very happy we no longer are passing sqldb around everywhere. I was able to run this branch in docker with no issue. My only question/suggestion is in genesis.go LN 225 regarding the if !exist, shouldn't we be handling an error there? Such as if exist {}?

**Comment by dbkbali** [comment]
@dutterbutter - The signature of block exists (LN 331 core/shyft_database_util.go) has been changed to return only a boolean - with error handling handled in BlockExists - ie. so in the event an error is returned when executing the db query it will  panic at line 340- BlockExists, which matches previous behavior except rather than triggering the panic in genesis.go by returning the full response for assessment of true, false or an error - genesis.go only has to deal with the true false scenario if there is no error.

File: core/db.go [hunk 1]
```diff
@@ -2,65 +2,142 @@ package core
...
+	if err1 != nil || err1 == sql.ErrNoRows {
+		fmt.Println(err1)
+	}
+	q2 := fmt.Sprintf(`DROP DATABASE IF EXISTS %s;`, dbName)
+	_, err2 := conn.Exec(q2)
+	if err2 != nil || err2 == sql.ErrNoRows {
+		fmt.Println(err2)
+	}
+	conn.Close()
+}
```
Comment (GregTheGreek) [inline core/db.go]: do we want to panic?

File: core/shyft_database_util.go [hunk 2]
```diff
@@ -332,21 +328,24 @@ func AccountExists(sqldb *sql.DB, addr string) (string, string, error) {
...
 	}
+	return res
 }
 
 //IsContract checks if toAddr is from a contract in Postgres Db
-func IsContract(sqldb *sql.DB, addr string) bool {
+func IsContract(addr string) bool {
+	sqldb, _ := DBConnection()
 	var isContract bool
 	sqlExistsStatement := `SELECT isContract from txs WHERE to_addr=($1)`
```
Comment (GregTheGreek) [inline core/shyft_database_util.go]: add `;` end of query

File: core/shyft_database_util.go [hunk 3]
```diff
@@ -358,9 +357,9 @@ func IsContract(sqldb *sql.DB, addr string) bool {
 	}
 }
 
-
 //UpdateAccount updates account in Postgres Db
-func UpdateAccount(sqldb *sql.DB, addr string, balance string, accountNonce string) {
+func UpdateAccount(addr string, balance string, accountNonce string) {
+	sqldb, _ := DBConnection()
 	updateSQLStatement := `UPDATE accounts SET balance = ($2), accountNonce = ($3) WHERE addr = ($1)`
```
Comment (GregTheGreek) [inline core/shyft_database_util.go]: here as well

**Review by GregTheGreek — CHANGES_REQUESTED** [review]
Just a few changes, add `;` on sql statements to remain consistent and potential panic. See comments.

**Review by GregTheGreek — APPROVED** [review]

**Pull request merged**